compiler/src/lntoamm/opcodes.ts

```diff
@@ -338,6 +338,8 @@ addopcodes({
   gtef32: [{ a: t('float32'), b: t('float32'), }, t('bool')],
   gtef64: [{ a: t('float64'), b: t('float64'), }, t('bool')],
   gtestr: [{ a: t('string'), b: t('string'), }, t('bool')],
+  httpget: [{ a: t('string')}, t('Result<any>')],
+  httppost: [{ a: t('string'), b: t('string')}, t('Result<any>')],
```

Comment:
This signature is *definitely* going to have to change once we add support for HTTP Headers.

Reply:
yes, I wanted to not have to think of it for now. Would we do the 3-arg trick and add a `Maybe<Array<KeyVal<Hashable, Any>>>`?

Reply:
> yes, I wanted to not have to think of it for now. Would we do the 3-arg trick and add a `Maybe<Array<KeyVal<Hashable, Any>>>`?

The 3-arg trick is only when there's no return value.

We have to do the trick done for the 3 and 4 arg reduce functions where we wrap the arguments into a user-defined type and then pass that down.

Reply:
Also, on the type up there, the `KeyVal` would be a `KeyVal<string, string>` since we know the types involved, and we can probably eliminate the `Maybe` by just using multiple dispatch to have a function that does or does not define a header payload and update the opcode's request object appropriately.

Reply:
Oh crap, how did I not noticed this? These opcodes are wrongly defined and `.getOr` was accidentally covering that up.

The return type should be `Result<string>` and there needs to be a line at the top of this file defining that type in opcode scope (around line 28)

Reply:
And we also need to make a ticket to figure out how `getOr` can override the return type instead of cause a compilation error here.

Reply:
ah, got it. Just pushed the fix, do you want to create the issue since you have more context?

Reply:
> ah, got it. Just pushed the fix, do you want to create the issue since you have more context?

Yes, though I discussed last night in chat before I passed out what went wrong here. :) The fix is unfortunately something we have to do after some other already planned work is done.